### PR TITLE
[DO NOT MERGE] Debug Nemotron NVFP4 MTP NaN on 0.5.13

### PR DESCRIPTION
**Pull request by mmangkad**
## Summary

Debug branch for the Nemotron NVFP4 MTP target-logits NaN on `release/v0.5.13`

---
### CI States

Latest PR Test (Base): :x: [Run #27296524557](https://github.com/sgl-project/sglang/actions/runs/27296524557)
Latest PR Test (Extra): :x: [Run #27296524503](https://github.com/sgl-project/sglang/actions/runs/27296524503)

### CODE REVIEW

**Comment by mmangkad** [comment]
/rerun-test test/registered/8-gpu-models/test_nvidia_nemotron_3_super_nightly.py test/registered/4-gpu-models/test_nvidia_nemotron_3_super_nvfp4.py

**Comment by mmangkad** [comment]
/rerun-test test/registered/4-gpu-models/test_nvidia_nemotron_3_super_nvfp4.py

**Comment by b8zhong** [comment]
@mmangkad I'm also debugging this issue. Here's what I currently found

> It's due to async assert env var enabled in https://github.com/sgl-project/sglang/pull/27461 (I think this NaN issue is pre-existing in Mamba2 kernels) (you can add some maybe detect nan in kernel outputs and find out that's the origin), and not caused by any particular recent PR that affects Nemotron/Mamba2
> Although, the accuracy and accept length is normal on GSM8K without this environment variable (and doesn't encounter any NaN in downstream sampling), thus I'm wondering if it's actually affectting anything right now (still should fix it later, but my opinion is it's OK to release...)

**Comment by mmangkad** [comment]
> @mmangkad I'm also debugging this issue. Here's what I currently found
> 
> > It's due to async assert env var enabled in #27461 (I think this NaN issue is pre-existing in Mamba2 kernels) (you can add some maybe detect nan in kernel outputs and find out that's the origin), and not caused by any particular recent PR that affects Nemotron/Mamba2
> > Although, the accuracy and accept length is normal on GSM8K without this environment variable (and doesn't encounter any NaN in downstream sampling), thus I'm wondering if it's actually affectting anything right now (still should fix it later, but my opinion is it's OK to release...)

@b8zhong yeah the env var is probably just revealing the NaN. Without the env var I don’t hit it at all, which is expected, but the weird part is that even with it enabled I can’t reproduce it consistently locally, though I have hit it a couple times. I think you’re right that this probably shouldn’t block release if accuracy and accept length look normal.

**Comment by b8zhong** [comment]
@mmangkad Usually, I think a critical NaN issue will propagate to sampling (and cause a server crash). But not sure why in this case it's fine...
